src/dom/node.ts

```diff
@@ -301,6 +301,32 @@ export class Node extends EventTarget {
     oldChild._replaceWith(newChild);
     return oldChild;
   }
+  normalize() {
+    let i = 0;
+    while (i < this.childNodes.length) {
+      const currentNode = this.childNodes[i];
+      if (currentNode.nodeType === NodeType.TEXT_NODE) {
+        let nextNode = this.childNodes[i + 1];
+
+        // Merge adjacent text nodes
+        while (nextNode && nextNode.nodeType === NodeType.TEXT_NODE) {
+          if (currentNode.nodeValue) {
```

Comment:
You don't need this `if`, adding an empty string would have the same effect

Reply:
Also, noticing that this `if` will drop the text from non-empty text nodes that follow empty text nodes

Reply:
Also, following the [standard algorithm outlined here](https://dom.spec.whatwg.org/#ref-for-dom-node-normalize%E2%91%A0) if any of the first text nodes found before others is empty then it's deleted and then the first non-empty text node will receive your `currentNode` treatment as you have here